FA.md

```diff
@@ -0,0 +1,45 @@
+NIP-FA
+======
+
+Kind-scoped follows
+-------------------
+
+`draft` `optional`
+
+This NIP defines kind `967`, a kind-scoped follow event.
+
+```jsonc
+{
+  "kind": 967,
+  "tags": [
+    ["p", "<followed-pubkey>", 'relay-url'],
+    ["k", "<some-kind>"],
+    ["k", "<some-other-kind>"]
+  ]
+}
+```
```

Comment:
should we encourage multiple kinds per follow events? isn't it better to have a different follow event for each kind such that they can be fetched and deleted independently?

Reply:
that will likely get too hairy; an app that supports 4 different kinds now needs to read/write 4x the number of events -- I think there's not enough benefit to forcing so much granularity and an app could just delete the event that has more kinds that it knows what to do and recreate with the kinds it doesn't understand (although even that might not even be necessary)

Reply:
but then we're back to lost followers again.

and also weird situations in which the same user will be following the same person multiple times because different apps have created different follow events, some bundled, some unbundled.

Reply:
not a big deal

Reply:
> but then we're back to lost followers again.

If an app misbehaves and deletes beyond what it should then it's a misbehaving app, but even within a misbehaving app, accidentally unfollowing someone across more use cases than the user intended to is FAR less problematic than unfollowing everyone across all use cases

> 
> and also weird situations in which the same user will be following the same person multiple times because different apps have created different follow events, some bundled, some unbundled.

An app that is interested in kind:20 will REQ for #k: "20", if it finds a follow it will use it, regardless of the fact that there are other `k`s in the follow event -- it will just ignore them, so it will show that person as followed.

Obviously when an app computes the `authors` filter based on the follow list it computed then it must make the list unique, but that's already pretty basic, same as if you have the same pubkey in a kind:3 which is also a thing

Reply:
ok, so apps must keep track of the event ids they're using to build their internal follow lists, right?

that means these same apps won't need `k` tags on the `kind:5` events in order to interpret an unfollow correctly (merging this with the other thread).

Reply:
if the unfollow is a delete and some other app deletes the follow event then they can't know the unfollow happened

they way I'm thinking about it is the app queries with

REQ with `{ "kinds": [5], "#k": ["967"], "since": last-known-unfollow }`